src/inventory.rs

```diff
@@ -23,17 +23,33 @@ pub trait Inventory {
             .collect()
     }
 
-    fn consume_one(&mut self, slot_id: SlotId) {
-        let mut slot = self.slot(slot_id).cloned();
-        if let Some(stack) = slot.as_mut() {
-            stack.set_count(stack.count() - 1);
-            let slot = if stack.count() == 0 {
+    /// Decreases the count for stack in the slot by amount. If there is not
+    /// enough items in the stack to perform the operation, then it will fail.
+    ///
+    /// Returns `Ok` if the stack had enough items, and the operation was
+    /// carried out. Otherwise, it returns `Err` if `amount > stack.count()`,
+    /// and no changes were made to the inventory.
+    #[allow(clippy::unnecessary_unwrap)]
+    fn consume(&mut self, slot_id: SlotId, amount: impl Into<u8>) -> Result<(), InventoryError> {
```

Comment:
I think the clones and unwraps can be avoided in this function.

Reply:
The clone cannot be avoided because we need `ItemStack`, not `&ItemStack`. Additionally, we can't use `if let Some(stack) = slot.as_mut()` because `stack` becomes `&mut ItemStack`, and we need to give `set_slot` an `Option<ItemStack>`.

That being said, moving the implementation of this function to the inventory types themselves would not really have this restriction. I'm thinking of consolidating `PlayerInventory` and `ConfigurableInventory` in another PR